src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -132,6 +134,29 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>, getSchedu
         createUpdateQueue(partialState);
       scheduleUpdate(fiber, updateQueue, LowPriority);
     },
+    enqueueCallback(instance, callback) {
+      const fiber = ReactInstanceMap.get(instance);
+      let updateQueue = fiber.updateQueue ?
+        fiber.updateQueue :
+        createUpdateQueue(null);
+      addCallbackToQueue(updateQueue, callback);
+
+      if (fiber.callbackList) {
+        // If this fiber was preempted and already has callbacks queued up,
+        // we need to make sure they are part of the new update queue
+        updateQueue = concatQueues(fiber.callbackList, updateQueue);
```

Comment:
Oops, this isn't right because this also adds the state updates the beginning of the queue! I believe this is unobservable unless there are side-effects inside state updaters (which there shouldn't be), but it's wasteful regardless.

Should instead combine the callbacks into a single callback, add to an empty node, and prepend it to the update queue.